apps/judicial-system/web/src/utils/hooks/useS3Upload/useS3Upload.ts

```diff
@@ -78,37 +78,34 @@ export const useUploadFiles = (files?: CaseFile[] | null) => {
     setUploadFiles(files?.map(mapCaseFileToUploadFile) ?? [])
   }, [files])
 
-  const allFilesUploaded = uploadFiles.every(
+  const allFilesDoneOrError = uploadFiles.every(
     (file) => file.status === 'done' || file.status === 'error',
   )
 
+  const someFilesError = uploadFiles.some((file) => file.status === 'error')
+
   const addUploadFile = (file: TUploadFile) =>
     setUploadFiles((previous) => [file, ...previous])
 
   const addUploadFiles = (
     files: File[],
     category?: CaseFileCategory,
+    status?: UploadFileStatus,
     policeCaseNumber?: string,
-    /**
-     * Use this to overwrite default file attributes, f.x.
-     * if you want to set a custom status or percent.
-     **/
-    overwriteDefaultFileAttributes?: Partial<TUploadFile>,
```

Comment:
This was done when implementing a new way of uploading files on some screens, f.x. when appealing a case, where files are not immediately uploaded, but you had to click "next" to actually upload the files. Does this PR address that scenario?

Reply:
Yes it does. I was mostly addressing exatly that scenario, but allowing parallel uploads.